python/controller.py

```diff
@@ -0,0 +1,32 @@
+import adafruit_rfm9x
+import busio
+import board
+import constants
+import rotaryio
+import time
+
+class RadioSender:
+  def __init__(self, animation_decoder, transreceiver):
+    self.animation_decoder = animation_decoder
+    # API doc: https://docs.circuitpython.org/projects/rfm9x/en/latest/api.html#adafruit_rfm9x.RFM9x
+    self.transreceiver  = transreceiver
+
+  def transmit_signal_callback(self, event, keypad_callback):
+    self.animation_decoder.query_rotary_position()
+    if keypad_callback(event) != constants.OFF or self.animation_decoder.did_update:
```

Comment:
1) might have to use getter for `did_update`

2) I think keypad_callback condition always returns true, can remove